core/common/src/main/java/alluxio/util/ShellUtils.java

```diff
@@ -234,13 +173,71 @@ public String toString() {
   }
 
   /**
-   * Static method to execute a shell command.
+   * Static method to execute a shell command. The StdErr is not returned.
+   * If execution failed
    *
    * @param cmd shell command to execute
    * @return the output of the executed command
+   * @throws IOException in various situations:
+   *  1. {@link ExitCodeException} when exit code is non-zero
+   *  2. when the executable is not valid, i.e. running ls in Windows
+   *  3. execution interrupted
+   *  4. other normal reasons for IOException
    */
   public static String execCommand(String... cmd) throws IOException {
-    return new Command(cmd).run();
+    return new ShellCommand(cmd).run();
```

Comment:
What are the existing uses of this? The new behavior now has stdout and stderr both as the returned output, and will that affect anything negatively?

Reply:
The behavior of this `run()` is not changed by this refactor. Only the `Command` class is renamed to `ShellCommand` (and now has a standalone class file) in this line. So there's no concern in the existing uses of this method.

For this `run()` method, stdout and stderr was always merged. And on non-zero exit code, there will be an exception thrown (which loses the output stdout and stderr). And this is exactly why I wanted to refactor and add another method preserving the output on error.